### PR TITLE
Use locale-independent camel-case transformation

### LINKED ISSUE #81: Pitfall using Turkish locale
Hi 👋 

thanks for this module – and all the other modules 🙏 

I recently ran [into a problem](https://github.com/peerigon/clockodo/issues/74) where my JSON objects were transformed into this:

```js
            "customersİd": 619336, // <-- notice the different capital i character
            "projectsİd": 631365,
            "usersİd": 62488,
```

Unfortunately, this didn't happen locally and during testing, it only happened in production in case the user was using the Turkish locale. There's already [an issue](https://github.com/sindresorhus/camelcase/issues/72) in the camelcase repo describing the problem.

I kind of understand that `toLocaleUpperCase()` is the default for the camelcase package since the context of the "to camelcase" transformation is not clear.

However, I think that `camelcase-keys` should use the locale-independent `toUpperCase()`. This is because I'm assuming that most people are using this module in a context where they want to transform API responses to camelCase in a predictable way. Using `toLocaleUpperCase()` poses the risk that API responses are transformed into a different shape on runtime. Also `toLocaleUpperCase("en")` doesn't really help since the locale might be unavailable and the host might fall back to the default locale.

I think this issue could be solved like this:

- Add an option to `camelcase` to use the locale-independent `toUpperCase()`
- Use that option as default in `camelcase-keys`

What do you think? 😀 I'd be willing to help out if that's ok.

### PR DESCRIPTION
**Pull request by jhnns**
Fixes #81

Not sure if we need to add tests here 😀

### CODE REVIEW

**Comment by sindresorhus** [comment]
A test would be good. Just to prevent any future accidental regression.

**Comment by jhnns** [comment]
I've added a test using a child process. I didn't want to mock String functions here since this would assume a specific implementation in `camelcase`

File: test.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import {execFile} from 'child_process';
 import test from 'ava';
 import camelcaseKeys from '.';
```
Comment (sindresorhus) [inline test.js:4]:
```suggestion
import {promising} from 'node:util';

const execFilePromise = promsify(execFile);
```

Comment (jhnns) [reply test.js:5]: Good idea 👍

I pushed a commit with this change. We can't use `node:util` on Node 12.x.

File: child-process-for-test.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+'use strict';
```
Comment (Richienb) [inline child-process-for-test.js:1]: This file should be moved into a `fixtures` directory

**Review by Richienb — CHANGES_REQUESTED** [review]

File: child-process-for-test.js (hunk 2 above)
Comment (jhnns) [reply child-process-for-test.js:1]: child-process-for-test.js is not a fixture to me. It's rather a test utility. What do you think @sindresorhus?

**Pull request merged**